climpred/bootstrap.py

```diff
@@ -234,6 +236,108 @@ def create_pseudo_members(control):
     )
 
 
+def bootstrap_uninit_pm_ensemble_from_control_cftime(init_pm, control):
+    """
+    Create a pseudo-ensemble from control run. Bootstrap random numbers for years to
+    construct an uninitialized ensemble from. This assumes a continous control
+    simulation without gaps.
+
+    Note:
+        Needed for block bootstrapping  a metric in perfect-model framework. Takes
+        random segments of length ``block_length`` from control based on ``dayofyear``
+        (and therefore assumes a constant climate control simulation) and rearranges
+        them into ensemble and member dimensions.
+
+    Args:
+        init_pm (xarray object): ensemble simulation.
+        control (xarray object): control simulation.
+
+    Returns:
+        uninit_pm (xarray object): uninitialized ensemble generated from control run.
+    """
+    check_lead_units_equal_control_time_stride(init_pm, control)
+    # short cut if annual leads
+    if init_pm.lead.attrs['units'] == 'years':
+        return bootstrap_by_stacking(init_pm, control)
+
+    block_length = init_pm.lead.size
+    freq = get_lead_cftime_shift_args(init_pm.lead.attrs['units'], block_length)[1]
+    nmember = init_pm.member.size
+    c_start_year = control.time.min().dt.year.astype('int')
+    # dont resample from years that control wont have timesteps for all leads
+    c_end_year = (
+        shift_cftime_singular(control.time.max(), -block_length, freq).dt.year.astype(
+            'int'
+        )
+        - 1
+    )
+
+    def sel_time(control, start_year_int, suitable_start_dates):
+        """Select time segments from control from ``suitable_start_dates`` based on
+        year ``start_year_int``."""
+        start_time = suitable_start_dates.time.sel(time=str(start_year_int))
+        end_time = shift_cftime_singular(start_time, block_length - 1, freq)
+        new = control.sel(time=slice(*start_time, *end_time))
+        new['time'] = init_pm.lead.values
+        return new
+
+    def create_pseudo_members(control, init):
+        """For every initialization take a different set of start years."""
+        startlist = np.random.randint(c_start_year, c_end_year, nmember)
+        suitable_start_dates = find_start_dates(init_pm, control, init)
```

Comment:
`init_pm` should be an argument to `create_pseudo_members` if it's being used here. Technically so should the above constants, but it's okay if they aren't if they are clearly constants with all capitals.

Reply:
`init_pm` is taken from the function above. want to avoid too many args.

Reply:
Same comment as before. I don't think it's good to reference a global variable inside a function without having it passed in. Maybe I'm wrong here. I'll ping Andrew on the gitter to see his thoughts. I don't think the number of args matters if it's all internal, i.e., the user isn't touching these functions.

Reply:
I can't find any best practices comments on this. I messaged @ahuang11 to see if he has any knowledge on this. I found this but no comments on what's good to do in this situation: https://www.python-course.eu/python3_global_vs_local_variables.php.

Reply:
For instance, why is `control` an input here while `init_pm` isn't? Only `init` is being varied at this level.

Reply:
I can also take out `control`